openmdao/core/driver.py

```diff
@@ -655,14 +665,37 @@ def add_constraint(self, name, lower=None, upper=None, equals=None,
         if isinstance(equals, np.ndarray):
             equals = equals.flatten()
 
-        # Scale the lower and upper values
-        if lower is not None:
-            lower = (lower + adder)*scaler
-        if upper is not None:
-            upper = (upper + adder)*scaler
+        # Scale the equality bounds
         if equals is not None:
             equals = (equals + adder)*scaler
 
+        # Scale the upper and lower bounds unless they are infinite.
+        if lower is None:
+            pass
+        elif isinstance(lower, np.ndarray):
+            lower = lower.flatten()
+            # lower is an array, only scale/offset indices > -inf_bound
+            lower = np.where( lower > -inf_bound, (lower + adder)*scaler, -inf_bound )
```

Comment:
For scalars it's not such an issue.  But the way pointer is posed, elements of a bound can often have different levels.  If I want to constrain the first element of an array and leave the others unbounded, I can't add the variable as a constraint twice with different indicies.

The correct way, as it is now, is to add the entire array with different bounds on different elements.